radicle-tracker/src/thread.rs

```diff
@@ -0,0 +1,162 @@
+use rose_tree::{NodeIndex, RoseTree, ROOT};
+use std::{collections::HashMap, hash};
+
+/// Laws:
+/// Thread::new(comment).first() === comment
+/// thread.reply(comment).delete(comment) === thread
+/// Thread::new(comment).delete(comment) === None
+/// Thread::new(comment).edit(f, comment) === Thread::new(f(comment).unwrap())
```

Comment:
Maybe clearer if you say `{delete,edit}(comment_path)`, because it looks like you're passing the actual comment in here.

Reply:
Ya they need updating! The comments and code in the `test` module below will be more accurate :grin: